src/librbd/ExclusiveLock.cc

```diff
@@ -206,19 +184,9 @@ void ExclusiveLock<I>::reacquire_lock(Context *on_reacquired) {
     Mutex::Locker locker(m_lock);
     assert(m_image_ctx.owner_lock.is_locked());
 
-    if (m_state == STATE_WAITING_FOR_REGISTER) {
```

Comment:
Doh -- forget this, since LockWather will disappear and ManagedLock will have very few interactions with a Watcher object. Will fix this at the ExclusiveLock class